SCClassLibrary/Common/Core/Document.sc

```diff
@@ -3,6 +3,33 @@ Document {
 	classvar <globalKeyDownAction, <globalKeyUpAction, <>initAction;
 	classvar <>autoRun = true;
 	classvar <asyncActions;
+
+	classvar <linkedIDE;
+	/* the IDE that will instantiate documents. It is that IDE's responsibility to implement
+	methods that may be invoked on it from the Document class:
+	  getQUuid
+	  newDocument
+	  setCurrentDocumentByQUuid
+	  open
+	  close
+	  setDocumentKeyDownEnabled
+	  setDocumentKeyUpEnabled
+	  setDocumentMouseDownEnabled
+	  setDocumentMouseUpEnabled
+	  setDocumentTextChangedEnabled
+	  setDocumentGlobalKeyDownEnabled
+	  setDocumentGlobalKeyUpEnabled
+	  setDocumentTitle
+	  setSelectionByQUuid
+	  setEditablebyQUuid
+	  setPromptsToSavebyQUuid
+	  removeDocUndoByQUuid
+
+	TODO: Add code to actually link `linkedIDE`
+	TODO: Replace all references to ScIDE with linkedIDE
+	*/
+
```

Comment:
yes, it would make it easier for developers of editors to know what is expected. Some things that are now in Document, esp. `allDocuments` and `currentDocument` are really quite important for the minimal set of programmatic functionality and should be required.

I am not completely sure if it might be better to keep the above two methods in Document, but I see your point.